src/ape/api/providers.py

```diff
@@ -1046,7 +1047,12 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
     def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         try:
-            txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            account = self.account_manager[txn.sender]
+            if isinstance(account, ImpersonatedAccount) and account.is_contract:
```

Comment:
Broader question, why should we only take this path if the impersonated account is a contract?